aioxmpp/entitycaps/service.py

```diff
@@ -334,50 +428,86 @@ def lookup_info(self, jid, keys):
 
     @aioxmpp.service.outbound_presence_filter
     def handle_outbound_presence(self, presence):
-        if (presence.type_ == aioxmpp.structs.PresenceType.AVAILABLE and
-                self.__current_keys):
-            self.logger.debug("injecting capabilities into outbound presence")
-            self.__115.put_keys(self.__current_keys, presence)
+        if (presence.type_ == aioxmpp.structs.PresenceType.AVAILABLE
+                and self.__active_hashsets):
+            current_hashset = self.__active_hashsets[-1]
+
+            try:
+                keys = current_hashset[self.__115]
+            except KeyError:
+                pass
+            else:
+                self.__115.put_keys(keys, presence)
+
+            try:
+                keys = current_hashset[self.__390]
+            except KeyError:
+                pass
+            else:
+                self.__390.put_keys(keys, presence)
 
         return presence
 
     @aioxmpp.service.inbound_presence_filter
     def handle_inbound_presence(self, presence):
-        keys = list(self.__115.extract_keys(presence))
-        lookup_task = asyncio.async(
-            self.lookup_info(presence.from_, keys)
-        )
-        self.disco_client.set_info_future(presence.from_, None, lookup_task)
+        keys = []
+
+        if self.xep390_support:
+            keys.extend(self.__390.extract_keys(presence))
+
+        if self.xep115_support:
+            keys.extend(self.__115.extract_keys(presence))
+
+        if keys:
+            lookup_task = asyncio.async(
```

Comment:
I don’t think that monkey patching is a good idea here. I was comfortable with monkey patching assert_not_called into Mock, but only because I could do that limited to the test suite.

Monkey patching in a dependency of ours is not something I feel we should be doing. (FWIW, we already do that in MLXC)